Gems/Atom/Tools/AtomToolsFramework/Code/Source/Util/Util.cpp

```diff
@@ -422,4 +425,64 @@ namespace AtomToolsFramework
         }
         return results;
     }
+
+    void AddRegisteredScriptToMenu(QMenu* menu, const AZStd::string& registryKey, const AZStd::vector<AZStd::string_view>& arguments)
+    {
+        // Map containing vectors of script file paths organized by category
+        using ScriptsSettingsMap = AZStd::map<AZStd::string, AZStd::vector<AZStd::string>>;
+
+        // Retrieve and iterate over all of the registered script settings to add them to the menu
+        for (const auto& [scriptCategoryName, scriptPathVec] : GetSettingsObject(registryKey, ScriptsSettingsMap()))
+        {
+            // Create a parent category menu group to contain all of the individual script menu actions.
+            QMenu* scriptCategoryMenu = menu;
+            if (!scriptCategoryName.empty())
+            {
+                scriptCategoryMenu = menu->findChild<QMenu*>(scriptCategoryName.c_str());
+                if (!scriptCategoryMenu)
+                {
+                    scriptCategoryMenu = menu->addMenu(scriptCategoryName.c_str());
+                }
+            }
+
+            // Create menu actions for executing the individual scripts.
+            for (AZStd::string scriptPath : scriptPathVec)
+            {
+                // Removing the alias for the path so that we can check for its existence and add it to the menu.
+                scriptPath = GetPathWithoutAlias(scriptPath);
+                if (QFile::exists(scriptPath.c_str()))
+                {
+                    // Use the file name instead of the full path as the display name for the menu action.
+                    AZStd::string filename;
+                    AZ::StringFunc::Path::GetFullFileName(scriptPath.c_str(), filename);
+
+                    scriptCategoryMenu->addAction(filename.c_str(), [scriptPath, arguments]() {
+                        // Delay execution of the script until the next frame.
+                        QTimer::singleShot(0, [scriptPath, arguments]() {
+                            AzToolsFramework::EditorPythonRunnerRequestBus::Broadcast(
+                                &AzToolsFramework::EditorPythonRunnerRequestBus::Events::ExecuteByFilenameWithArgs,
+                                scriptPath,
```

Comment:
Does this one not need to cast to UTF8?

Reply:
Thank you for reviewing. The first one is AZStd::string. The second one is a QString that came directly from the QT function that needs to be converted.